app/com/mohiva/play/compressor/CompressorFilter.scala

```diff
@@ -88,17 +88,31 @@ abstract class CompressorFilter[C <: Compressor] extends Filter {
 
     if (isCompressible(result)) {
       result.body match {
-        case HttpEntity.Strict(data, contentType) =>
-          Future.successful(Result(result.header, HttpEntity.Strict(ByteString(compress(data)), contentType)))
-        case HttpEntity.Streamed(data, _, contentType) =>
-          data.toMat(Sink.fold(ByteString())(_ ++ _))(Keep.right).run() map { bytes =>
+
+        case body0: HttpEntity.Strict =>
+          val result2 = result.copy(
+            body = body0.copy(
+              data = ByteString(compress(body0.data))
+            )
+          )
+          Future.successful(result2)
+
+        case body0: HttpEntity.Streamed =>
+          for {
+            bytes <- body0.data
+              .toMat(Sink.fold(ByteString())(_ ++ _))(Keep.right).run()
+          } yield {
             val compressed = compress(bytes)
             val length = compressed.length
-            Result(
-              result.header.copy(headers = result.header.headers),
-              HttpEntity.Streamed(Source.single(ByteString(compressed)), Some(length.toLong), result.body.contentType)
+            result.copy(
+              //header = result.header.copy(headers = result.header.headers),
```

Comment:
Hmm, this code previously was UNcommented. But i think, this code makes no sense.
May be, something was or invalid missing here?